lgsm/functions/command_start.sh

```diff
@@ -8,6 +8,7 @@
 commandname="START"
 commandaction="Starting"
 functionselfname="$(basename "$(readlink -f "${BASH_SOURCE[0]}")")"
+add_ts="gawk '{ print strftime(\\\"[$logtsformat]\\\"), \\\$0 }'"
```

Comment:
Suggesting to use date instead of gawk:
`printf '[%s] %s\n' "$(date "+$logtsformat")" "hello world"`
or `echo "asd" | xargs printf '[%s] %s\n' "$(date "+$format")"`
provides the same output as
echo "hello world" | gawk '{ print strftime("[%Y-%m-%d %H:%M:%S]"), $0}'
and we wouldnt need an additional dependency

Reply:
I had a hard time escaping that command to be able to use it in my exec . If you managed to do it I'd be happy to use it.

Reply:
Simple explanation:
bash knows mostly strings, 
`"interpret special characters >once< before passing this to something"`
`'dont interpret special characters before passing'`

Therefore:
`echo "$UID" '$UID' # whitespace in between = two strings = two args for echo`  1st is checked for special characters and after that added as arg to echo, 2nd not
If you dont want this behavior use single quotes.

If you need to mix it, you can combine it.
`echo "$UID"'$UID' # no whitespace in between = one string, first part is interpreted before`
or you can escape with \
`echo "\$UID" '$UID' # output is identical`
if you do want to output \ between "", you need to escape it as well:
`echo "\\" '\' # but not if you use single quotes`
What if you want to use double or single quotes?
`echo "\"" '"'` or `echo "'"`
You can also escape whitespaces, so this is one string
`foo=hello\ world` but please dont do so

In your command: `add_ts="gawk '{ print strftime(\\\"[$logtsformat]\\\"), \\\$0 }'"` `add_ts=".."..".."` you used the "interpret special characters way" but you wanted to add double quotes to it, so you needed to escape it once. `add_ts="..\"..\".."` but why are the additional \ needed?
```bash
foo=hello
bar=\$foo
echo "$bar" # prints $foo because interpreted once, even if special symbol is in
eval echo "$bar" # prints hello but why?
# calls eval "echo" '$foo'
# eval interpreted once more and executes "echo" "hello"
#
# if you want to prevent it ( and how you did )
bar=\\\$foo
eval echo "$bar" # prints now $foo instead of hello
# thats what happened to you
```

Sry was more motivated to clarify this for you than fixing the issue.

Reply:
I understand all that, but I need it to process the variable `$logtsformat` inside of `add_ts`. Which means I have:


```bash
add_ts=gawk '{ print strftime(\"[%Y-%m-%d %H:%M:%S]\"), \$0 }'
``` 

Then in L96 the line will become:

```bash
tmux pipe-pane -o -t "${sessionname}" "exec bash -c \"cat | gawk '{ print strftime(\"[%Y-%m-%d %H:%M:%S]\"), \$0 }'\" >> '${consolelog}'"
```

I can put everything manually in that line, but we'll have the some logic repeated in multiple steps and it'll make the code less readable and harder to maintain.